backends/postgres/postgres_backend_test.go

```diff
@@ -410,6 +410,71 @@ func TestCron(t *testing.T) {
 	})
 }
 
+func TestMultipleCronNodes(t *testing.T) {
+	jobsProcessed := sync.Map{}
+	const numberOfWorkers = 4
+	const cron = "* * * * * *"
+	connString := os.Getenv("TEST_DATABASE_URL")
+	if connString == "" {
+		t.Skip("Skipping: TEST_DATABASE_URL not set")
+		return
+	}
+	t.Cleanup(func() {
+		flushDB()
+	})
+
+	workers := make([]neoq.Neoq, numberOfWorkers)
+	var jobsCompleted uint32
+	var duplicateJobs uint32
+	for i := 0; i < numberOfWorkers; i++ {
+		ctx := context.TODO()
+		nq, err := neoq.New(ctx, neoq.WithBackend(postgres.Backend), postgres.WithConnectionString(connString))
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Cleanup(func() {
+			nq.Shutdown(ctx)
+		})
+		h := handler.NewPeriodic(func(ctx context.Context) (err error) {
+			job, err := jobs.FromContext(ctx)
+			if err != nil {
+				t.Fatalf("failed to extract job details from context: %+v", err)
+				return nil
+			}
+			_, exists := jobsProcessed.LoadOrStore(job.ID, "foo")
+			if exists {
+				t.Fatalf("job (%d) has already been processed by another worker!", job.ID)
+				atomic.AddUint32(&duplicateJobs, 1)
+				return nil
+			}
+			atomic.AddUint32(&jobsCompleted, 1)
+			return
+		})
+
+		h.WithOptions(
+			handler.JobTimeout(500*time.Millisecond),
+			handler.Concurrency(1),
+		)
+
+		err = nq.StartCron(ctx, cron, h)
+		if err != nil {
+			t.Error(err)
+		}
+
+		workers[i] = nq
+	}
+
+	// allow time for listener to start and for at least one job to process
+	time.Sleep(2 * time.Second)
```

Comment:
We could really wait as long as we want here, the longer it runs the more crons get processed and the more we assert that we don't have duplicates. But I don't want to introduce a long running test for the sake of having a long running test. Up to you if you want to have this run longer?

Reply:
It will be preferable to bump `numberOfWorkers` up so that we can test more enqueueing events in the smallest amount of time possible. Is 50 realistic? We may bump into Github's 100 connections per PG server limit at some point; there's a way to change that limit, but it's a little ugly.

I would also drop that sleep time down to something like `1050 * time.Millisecond` and give the `Duration` a name like `SlightlyMoreThanOneSecond`.

Reply:
Added!